File: vaku2/path_move.go
package vaku2

import (
	"errors"
)

var (
	ErrPathMove = errors.New("path move")
)

// PathMove moves data at a source path to a destination path (copy + delete). Client must
// have been initialized using WithDstClient() when moving across vault servers.
func (c *Client) PathMove(src, dst string) error {
	err := c.PathCopy(src, dst)
	if err != nil {
		return newWrapErr("copy "+src+" to "+dst, ErrPathMove, err)
	}

	err = c.PathDelete(src)
	if err != nil {
		return newWrapErr("delete "+dst, ErrPathMove, err)
	}

	return nil
}

File: src/api/path_move.rs
//! `PathMove` — copy then delete source.

use crate::api::client::Client;
use crate::api::error::{Error, ErrorKind};

impl Client {
    /// Move (copy + delete) the current version from `src` to `dst`.
    pub async fn path_move(&self, src: &str, dst: &str) -> Result<(), Error> {
        self.path_copy(src, dst)
            .await
            .map_err(|e| Error::wrap("", ErrorKind::PathMove, Some(Box::new(e))))?;
        self.path_delete(src)
            .await
            .map_err(|e| Error::wrap(dst, ErrorKind::PathMove, Some(Box::new(e))))
    }
}
